src/bluealsa.conf.in

```diff
@@ -7,7 +7,7 @@
 
   <!-- ../system.conf have denied everything, so we just punch some holes -->
 
-  <policy user="root">
+  <policy user="@bluealsauser@">
```

Comment:
I'm just wondering whether we should replace root policy with selected user, or maybe we should add new policy for selected user.... On my distro I can see that the config for pulseaudio indeed does not contain root user. However, pulseaudio runs in default context (which is also permitted), not as a global service.... hmm...

And there is also one other thing. On my Debian installation, the bluetooth.conf has got something like this:
```xml
  <policy user="root">
    <allow own="org.bluez"/>
    <allow send_destination="org.bluez"/>
.....
  </policy>
  <policy group="bluetooth">
    <allow send_destination="org.bluez"/>
  </policy>
  <policy context="default">
    <allow send_destination="org.bluez"/>
  </policy>
```

So, if bluealsa will run as non-root, it will have to be in the bluetooth group, or run by the logged-in user. @borine have you checked that setup with non-root bluealsa is possible with your distro?

Reply:
Since bluez commit https://github.com/bluez/bluez/commit/3ef0ce954b66fdf45538a6cdc629f3dac6642832 (release 5.51 sept 2019) every user can send to `org.bluez`, effectively making the `bluetooth` group redundant (unless the distribution has added further restrictions)

So perhaps, to cover both pre- and post- 5.51 bluez installations, we should also add
`<allow send_destination="org.bluez">` to the bluealsa user policy ?

Reply:
I think it is sufficient for the install to ensure that the configured bluetooth service user can own "org.bluealsa". There is no technical reason why root would also need that permission. Distributions and admins are free add any extra permissions they choose.

I've now added the extra `<allow>` permission explicitly for older bluez releases where the default context had a `<deny>` rule for sending to org.bluez

I think that restricting org.bluealsa method calls to members of the `audio` group has always been sufficient, but I suppose it could be extended to all users if you prefer. My instinct is to select the least permissive alternative, but don't see any immediate objection to relaxing that rule if that is your preference.

Reply:
OK, my bad, I mistook `context="default"` for `at_console="true"`.

> There is no technical reason why root would also need that permission.

OK, I agree. Let's keep it like it is right now.

> I've now added the extra <allow> permission explicitly for older bluez releases 

I'm not sure whether it is OK to change setup for bluez.org in bluealsa.conf.... Maybe we should only add info in the manual/wiki, that using non-root user with bluez < 5.51 requires extra setup in bluetooth.conf? Maybe someone else has got some thoughts on that?

> I think that restricting org.bluealsa method calls to members of the audio group has always been sufficient

OK, keep it like it is.

Reply:
> I'm not sure whether it is OK to change setup for bluez.org in bluealsa.conf....

Hmm, its certainly permitted by the dbus docs, but it is quite possible that this would be considered "bad etiquette" by others. I don't know what the protocol is here, and I'm not known for my sensitivity to such matters ! :laughing: 

The only alternative for older bluez releases is to add the bluealsa user to the bluetooth group, so if you prefer I could add some text to the configure help (incidentally, the configure help might also mention that the bluealsa-aplay user must be a member of the audio group). Would you prefer that approach?

Reply:
.., If there are any distribution managers/packagers reading this, would you prefer:
a) bluealsa unilaterally grants itself permission to access org.bluez
b) the distribution grants that permission
c) the distribution adds its bluealsa user to the bluetooth group

... or perhaps just upgrade bluez!

Reply:
Yes, I think that the second approach will be better. I think that we shall not modify default setup of others, especially, if this is required only for older versions of bluez.